### DISCUSSION TITLE
Stub browser API when executed during SSR/SSG

### DISCUSSION THREAD

**Discussion by or2e** [nuxt/nuxt#17709] [Questions] [answered]
Hi ;)

How i can stub ResizeObserver only executed during SSR/SSG builds?

For example, this component used ResizeObserver: https://github.com/DerYeger/vue-masonry-wall/blob/master/src/masonry-wall.vue#L118

stub example
` global.ResizeObserver = class ResizeObserver {
      public constructor() {}
      public disconnect() {}
      public observe() {}
      public unobserve() {}
    }`

**Comment by atinux** [comment, marked as answer]
I think you should open a PR to the library in order to make it SSR compatible directly 😊

**Reply by or2e** [reply]
Thanks for answer )

The author of the library declared support for SSR/SSG by adding a stub:
https://github.com/DerYeger/vue-masonry-wall/issues/24#issuecomment-1073951038
